packages/modern-poalim-scraper/src/scrapers/cal.ts

```diff
@@ -0,0 +1,304 @@
+import { Frame, Page } from 'puppeteer';
+import { subYears } from 'date-fns';
+import { waitUntil } from '../helpers/waiting.js';
+import type { CalGetCardTransactionsDetailsResponse, CalTransaction } from './types/cal/get-card-transactions-details.js';
+import { fetchPostWithinPage } from '../utils/fetch.js';
+
+const LOGIN_URL = 'https://www.cal-online.co.il/';
+const TRANSACTIONS_REQUEST_ENDPOINT = 'https://api.cal-online.co.il/Transactions/api/transactionsDetails/getCardTransactionsDetails';
+
+const last4DigitsToCardIds: Record<string, string> = {}; // last4Digits -> cardId
+
+function getCardId(last4Digits: string) {
+  return last4DigitsToCardIds[last4Digits];
+}
+
+export async function cal(page: Page, credentials: CalCredentials, options: CalOptions = {}) {
+  await login(credentials, page);
+
+  const authToken = await getAuthorizationHeader(page);
+  const xSiteId = await getXSiteId();
+
+  const cards = await getCards(page);
+  for (const card of cards) {
+    last4DigitsToCardIds[card.last4Digits] = card.cardUniqueId;
+  }
+
+  return {
+    getMonthTransactions: async (last4Digits: string, month: Date) => {
+      const cardId = getCardId(last4Digits);
+      if (!cardId) 
+        throw new Error(`Card ID not found for last 4 digits: ${last4Digits}`);
+      return fetchMonthCompletedTransactions(
+        page,
+        cardId,
+        authToken,
+        xSiteId,
+        month
+      );
+    },
+    getTransactions: async () => {
+      return fetchTransactions(page, options, authToken, xSiteId);
+    },
+  }
+}
+
+async function login(credentials: CalCredentials, page: Page) {
+  await page.goto(LOGIN_URL);
+  
+  // Wait for and click the login button
+  await page.waitForSelector('#ccLoginDesktopBtn', { visible: true });
+  await page.click('#ccLoginDesktopBtn');
+  
+  // Get the login frame using the proper frame detection
+  const frame = await getLoginFrame(page);
+
+  await sleep(1000);
```

Comment:
are these necessary in case you are waiting for an element?

Reply:
I can replace all the sleeps with waitFor

Reply:
Done

Reply:
I ended up putting one of the sleeps back. Couldn't find a good way to not have it. Spent too long trying to make it work without the sleep :(

Reply:
I could keep trying but doesn't seem like a productive use of time